Tutorials/i2c.md

```diff
@@ -0,0 +1,110 @@
+
+Let us get the accelerometer values using the I2C protocol. We will be using the [The Tessel Accelerometer](https://www.seeedstudio.com/Tessel-Accelerometer-Module-p-2223.html)
+
+[Refer to the datasheet for the accelerometer module to follow along](http://www.nxp.com/docs/en/data-sheet/MMA8452Q.pdf)
+
+![Fritzing Diagram](http://i.imgur.com/zK4U4S3.png)
+
+```js
+var tessel = require('tessel');
+
+// Connect to device
+var port = tessel.port.A; // Use the SCL/SDA pins of Port A
+
+// This address of the Slave has been taken from https://github.com/tessel/accel-mma84/blob/master/index.js#L15
+// It can also be found in Table 10 - I2C device address sequence on page number 17 of the datasheet. 
+var slaveAddress = 0x1D; // Specific to device
+
+var i2c = new port.I2C(slaveAddress); // Initialize I2C communication
+
+// Details of I2C read and I2C transfer
+var numBytesToRead = 1; // Read back this number of bytes
+
+// Read data over I2C using i2c.read
+i2c.read(numBytesToRead, function (error, dataReceived) {
+  // Print data received (buffer of hex values)
+  console.log('Buffer returned by I2C slave device ('+slaveAddress.toString(16)+'):', dataReceived);
+
+});
+
+// Read/Receive data over I2C using i2c.transfer
+// 0x0D is the WHO_AM_I Register which sends back an acknowledgement to the master for starting the communication
+// Information about 0x0D can also be found in Table 11 - Register address map on page number 19 of the datasheet.
+
+i2c.transfer(new Buffer([0x0D]), numBytesToRead, function (error, dataReceived) {
+    // Print data received (buffer of hex values)
+    // The returned buffer from the I2C slave device should be [0x2A] as specified in the register address map on page number 19 of the MMA84 datasheet
+  console.log('Buffer returned by I2C slave device ('+slaveAddress.toString(16)+'):', dataReceived);
+
+});
+
+/*
+
+  In the i2c.transfer function, the dataReceived consists of all the 6 bytes of data (which is 48 bits of data) : 2 bytes each for the x, y, and z values.
+  This can be seen on page 20 and 21 of the accelerometer datasheet.
+  Each of the x, y, and z acceleration values have two registers associated with them for storing the 12 bit long sample.
+  The first 8 bits are stored in their respective OUT_MSB registers. These are the Most Significant first 8 bits.
+  The next 4 bits are stored in their respective OUT_LSB registers. The remaining 4 bits are occupied
+  by 0s. These lower 4 bits (bits 0, 1, 2, and 3 of the OUT_LSB registers) are redundant bits which are not required. The OUT_LSB and OUT_MSB store the 2's complement form of the coordinates.
+
+  The organization of the registers can be seen in the datasheet inside section 6.1 (Data Registers), page number - 21
+  The register address for OUT_X_MSB is 0x01. This can be found at Page 19 of https://www.nxp.com/docs/en/data-sheet/MMA8452Q.pdf
```

Comment:
Useful to note that this chip lets you read registers sequentially, so that's why we only need to specify the first address